common.rkt

```diff
@@ -150,17 +145,14 @@
                     ((var? t)  (set! index (+ 1 index))
                                (state (extend-sub t (reified-index index) (state-sub st)) (state-diseq st) (state-types st)))
                     (else      st)))))
-    (cond
-      ((and (null? (state-diseq st)) (null? (state-types st))) (walk* tm results))
-      ((null? (state-types st)) (Ans 
-                                  (walk* tm results)
-                                  (list (walk* (cons '=/= (map pretty-diseq (state-diseq st))) results))))
-      ((null? (state-diseq st)) (Ans
-                                  (walk* tm results)
-                                  (walk* (map pretty-types (state-types st)) results)))
-      (else (Ans (walk* tm results) 
-                (append (walk* (map pretty-types (state-types st)) results) 
-                        (list (walk* (cons '=/= (map pretty-diseq (state-diseq st))) results) )))))))
+    (let* ((walked-sub (walk* tm results))
+          (diseq-null (null? (state-diseq st)))
+          (types-null (null? (state-types st)))
```

Comment:
Like predicates, boolean names are nicer if they end with a `?`.  So here we would name these `diseq-null?` and `types-null?`.

However, see a later comment where I suggest getting rid of these altogether.